Automattic/woocommerce-payments
# Use WordPress `CardBody` and `CardDivider` components for standard padding in deposits card

Fixes #5954 

**Merge note**: This PR adjusts markup on two cards under heavy development (lots of open PRs). We've decided to hold off merging this PR until the current open PRs are merged, to avoid conflicts delaying the core feature work.

#### Changes proposed in this Pull Request

This PR refactors our two new cards (`DepositsOverview` and `DepositsInformationOverview` (aka Balances card?) to make full use of [card-related components from WordPress core](https://developer.wordpress.org/block-editor/reference-guides/components/card/). 

Previously, our card child components were individually adding the card padding.

Here's a screenshot of the new `DepositsOverview` card from this branch:

![Screenshot 2023-04-04 at 12 38 43 PM](https://user-images.githubusercontent.com/4167300/229656726-ae5175ce-a69a-4bba-9f76-5795b8cba560.png)

Specific changes:

- `DepositsInformationOverview` now uses `CardBody`. No other changes – this didn't break the existing styling. 
- `NextDepositDetails`
  - A `CardBody` for each child component (header, table header, table row). 
  - A `CardDivider` for the table header divider.
  - Removed some padding and custom divider style rules (now supplied by WP core styles).
- `DepositsOverview` has a class applied to the outer container `Card`. This is so can override (reset) `CardBody` padding as needed in the stacked `CardBody`s.

#### Testing instructions

* Set up a Woo store and onboard WCPay. Make some purchases so you have a balance to deposit.
* Enable the [deposits improvements feature flag](https://github.com/Automattic/woocommerce-payments/pull/5763): set `_wcpay_feature_simplify_deposits_ui` wp_option to `.
* Go to `Dashboard > Payments > Overview`.  
* Review the two deposits-related cards and ensure layout and look is correct and matches designs. 
* Test any interesting relevant deposits scenarios & ensure the cards are correct.

-------------------


**Post merge**
